### ISSUE TITLE
Add .Status field to CRs

### ISSUE THREAD

**Issue by paulfantom** [coreos/prometheus-operator#3335]
**What is missing?**

Currently `.Status` field for objects used by prometheus-operator (PodMonitors, ServiceMonitors, Prometheus, PrometheusRule...) is not populated by operator. It would be good to include some information in this field. Some examples:

1) For Pod/ServiceMonitor it could have number of discovered targets. 
2) For Prometheus it could have some `/status` information from prometheus instance.
3) PrometheusRule could show rule group names and rule count.

**Why do we need it?**

To increase visibility and ease debugging process.

**Environment**

* Prometheus Operator version: ALL

**Anything else we need to know?**:

**Comment by brancz** [comment]
Agreed that this would be amazing, I've thought about this for a long time there are two problems involved here that are not necessarily trivial to answer:

1) how to perform the communication between the prometheus operator and Prometheus pods securely (for real status and targets this is required)
2) deduplication of rules, targets, etc., this has been solved elsewhere so I'm not entirely convinced it has to be solved yet again, but I see the appeal of making it "kubernetes native"

As an added benefit of 1) if we had this, we could have instantaneous updates of ServiceMonitors/PodMonitors/PrometheusRules.

**Comment by paulfantom** [comment]
> how to perform the communication between the prometheus operator and Prometheus pods securely (for real status and targets this is required)

This part should be simpler now since Prometheus 2.24.0 has built-in TLS handling (and we already have an issue to enable it - https://github.com/prometheus-operator/prometheus-operator/issues/3782)

**Comment by simonpasquier** [comment]
I've submitted https://github.com/prometheus-operator/prometheus-operator/pull/4575 where we can discuss how the status subresources should look like.

**Comment by aerosouund** [comment]
I'm highly interested in working on this project for GSoC 24.
As someone who habitually works with kubernetes and prometheus i'd love to get the chance to contribute to it. I'll currently be looking at this [PR](https://github.com/prometheus-operator/prometheus-operator/pull/4575) to understand how this discussion concluded earlier and what are the problems to tackle and contact you if i have any questions

**Comment by simonpasquier** [comment]
I've captured my thoughts in https://gist.github.com/simonpasquier/78dda3dbda52485cc4da05370088c3d9. I've got no time right to submit this as a proposal but feel free to pick it up if interested.

**Comment by hanshal101** [comment]
Hey is this issue still relevent?

**Comment by lterrac** [comment]
Hi all, I would like to write a proposal for GSOC '25 for this project. I wrote k8s operators for a couple of years, so I think this might be a good starting point to enter the open source world 😄  Beside all the content available in this issue, is there any other relevant resource I should look at? Do you have any suggestion to write an effective proposal for this project? 😄 
Thanks in advance for your time!

**Comment by heliapb** [comment]
Hi @lterrac you can take a look at https://github.com/prometheus-operator/community/blob/main/mentoring/gsoc/2025/project_ideas.md for GSoC 25 and also there's going to happen today a extra office hours if you have questions

**Comment by lterrac** [comment]
Hi @heliapb thanks the pointer 😄 unfortunately I couldn't make it to today's meeting. I am starting to look into the codebase and also started reading the last thoughts on this issue (like the gist posted a few messages above). I am going to prepare the application and send it as soon as I can.
I just want to point out that I am currently employed, so I would take this project on my spare time. I haven't seen any restriction in the GSOC website about this, but I am just stating it for the sake of transparency. Hope this will not be an issue 😄

**Comment by heliapb** [comment]
Hi @lterrac ofc is for any new opens source contributors as per the document https://developers.google.com/open-source/gsoc/faq. Also any question drop on the K8s slack channel

**Comment by lterrac** [comment]
Will do! Many thanks 😄

**Comment by shams03** [comment]
Hi @heliapb I am currently working on Go and I have learnt k8s in the past , although not used it much . I would love to be considered for Gsoc '25 , currently I've been reading the codebase for sometime . 
Will update on my thoughts and ideas .
Thank You for your time
